src/snowflake/snowpark/relational_grouped_dataframe.py

```diff
@@ -365,6 +366,103 @@ def end_partition(self, pdf: pandas.DataFrame) -> pandas.DataFrame:
+            ... (1, 'A', 3000, 'FEB'),
+            ... (2, 'B', 200, 'FEB') ''').collect()
+            >>> df = session.table("monthly_sales")
+            >>> df.group_by("empid").pivot("month", ['JAN', 'FEB']).sum("amount").sort(df["empid"]).show()
+            -------------------------------
+            |"EMPID"  |"'JAN'"  |"'FEB'"  |
+            -------------------------------
+            |1        |10400    |8000     |
+            |2        |39500    |200      |
+            -------------------------------
+            <BLANKLINE>
+
+        Example::
+
+            >>> create_result = session.sql('''create or replace temp table monthly_sales(empid int, team text, amount int, month text)
+            ... as select * from values
+            ... (1, 'A', 10000, 'JAN'),
+            ... (1, 'B', 400, 'JAN'),
+            ... (2, 'A', 4500, 'JAN'),
+            ... (2, 'A', 35000, 'JAN'),
+            ... (1, 'B', 5000, 'FEB'),
+            ... (1, 'A', 3000, 'FEB'),
+            ... (2, 'B', 200, 'FEB') ''').collect()
+            >>> df = session.table("monthly_sales")
+            >>> df.group_by("empid").pivot("month").sum("amount").sort(df["empid"]).show()
+            -------------------------------
+            |"EMPID"  |"'JAN'"  |"'FEB'"  |
+            -------------------------------
+            |1        |10400    |8000     |
+            |2        |39500    |200      |
+            -------------------------------
+            <BLANKLINE>
+
+        Example::
+
+            >>> create_result = session.sql('''create or replace temp table monthly_sales(empid int, team text, amount int, month text)
+            ... as select * from values
+            ... (1, 'A', 10000, 'JAN'),
+            ... (1, 'B', 400, 'JAN'),
+            ... (2, 'A', 4500, 'JAN'),
+            ... (2, 'A', 35000, 'JAN'),
+            ... (1, 'B', 5000, 'FEB'),
+            ... (1, 'A', 3000, 'FEB'),
+            ... (2, 'B', 200, 'FEB') ''').collect()
+            >>> df = session.table("monthly_sales")
+            >>> df.group_by(["empid", "team"]).pivot("month", ['JAN', 'FEB']).sum("amount").sort("empid", "team").show()
+            ----------------------------------------
+            |"EMPID"  |"TEAM"  |"'JAN'"  |"'FEB'"  |
+            ----------------------------------------
+            |1        |A       |10000    |3000     |
+            |1        |B       |400      |5000     |
+            |2        |A       |39500    |NULL     |
+            |2        |B       |NULL     |200      |
+            ----------------------------------------
+            <BLANKLINE>
+        """
+        pc = self._df._convert_cols_to_exprs(
+            "RelationalGroupedDataFrame.pivot()", pivot_col
+        )
+        if values is None:
+            distinct_values = (
+                self._df.select(pivot_col).distinct()._internal_collect_with_tag()
+            )
+            value_exprs = [Literal(v[0]) for v in distinct_values]
```

Comment:
should we add this in `Dataframe.pivot` as well with a documentation update saying this method is not recommended?